pkg/controller/master/pvc/pvc_controller.go

```diff
@@ -1,21 +1,62 @@
 package pvc
 
 import (
+	"fmt"
+
 	ctlcorev1 "github.com/rancher/wrangler/v3/pkg/generated/controllers/core/v1"
 	"github.com/sirupsen/logrus"
 	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
 	cdicommon "kubevirt.io/containerized-data-importer/pkg/controller/common"
 
 	ctlcdiv1 "github.com/harvester/harvester/pkg/generated/controllers/cdi.kubevirt.io/v1beta1"
+	"github.com/harvester/harvester/pkg/util"
 )
 
 // pvcHandler used to separate the PVC from the DataVolume
 type pvcHandler struct {
-	dataVolumeClient ctlcdiv1.DataVolumeClient
-	pvcClient        ctlcorev1.PersistentVolumeClaimClient
-	pvcController    ctlcorev1.PersistentVolumeClaimController
+	volImportSourceClient ctlcdiv1.VolumeImportSourceClient
+	dataVolumeClient      ctlcdiv1.DataVolumeClient
+	pvcClient             ctlcorev1.PersistentVolumeClaimClient
+	pvcController         ctlcorev1.PersistentVolumeClaimController
+}
+
+func (h *pvcHandler) createFilesystemBlankSource(_ string, pvc *corev1.PersistentVolumeClaim) (*corev1.PersistentVolumeClaim, error) {
+	if pvc == nil || pvc.Spec.VolumeMode == nil || *pvc.Spec.VolumeMode != corev1.PersistentVolumeFilesystem {
+		return pvc, nil
+	}
+
+	annotations := pvc.GetAnnotations()
+	if v, find := annotations[util.AnnotationVolForVM]; !find || v != "true" {
+		logrus.Debugf("PVC %s/%s is not for VM, skip checking filesystem blank", pvc.Namespace, pvc.Name)
+		return pvc, nil
+	}
+
+	targetPopulatorName := fmt.Sprintf("filesystem-blank-source-%s", pvc.Namespace)
```

Comment:
The namespace suffix is unnecessary since `VolumeImportSource` is already scoped to the namespace. That will make sure we don't violate the naming chars limitation per @w13915984028.

Reply:
udpated